ibis/spark/client.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import
+
+import pyspark
+from pyspark import SparkConf, SparkContext
+from pyspark.sql import SparkSession
+from pyspark.sql.types import (BooleanType, NullType, ArrayType,
+                               StringType, BinaryType, DateType,
+                               TimestampType, ShortType, IntegerType,
+                               LongType, FloatType, DoubleType,
+                               DecimalType, StructType)
+
+import ibis.client as client
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+
+from ibis.compat import parse_version
+
+
+_ibis_dtypes = {
+    dt.Boolean: BooleanType,
+    dt.Null: NullType,
+    dt.Array: ArrayType,
+    dt.String: StringType,
+    dt.Binary: BinaryType,
+    dt.Category: 'category',  # FIXME
+    dt.Date: DateType,
+    dt.Time: TimestampType,
+    dt.Timestamp: TimestampType,
+    dt.Int8: ShortType,
+    dt.Int16: ShortType,
+    dt.Int32: IntegerType,
+    dt.Int64: LongType,
+    dt.UInt8: IntegerType,  # FIXME
+    dt.UInt16: IntegerType,  # FIXME
+    dt.UInt32: LongType,
+    dt.UInt64: LongType,   # FIXME
+    dt.Float32: FloatType,
+    dt.Float64: DoubleType,
+    dt.Decimal: DecimalType,
+    dt.Struct: StructType
+}
+
+
+def ibis_dtype_to_spark(ibis_dtype):
+    """Convert ibis dtype to the pandas / numpy alternative"""
+    return _ibis_dtypes[type(ibis_dtype)]
+
+
+def ibis_schema_to_spark(schema):
+    return list(zip(schema.names, map(ibis_dtype_to_spark,
+                                      schema.types)))
+
+
+class SparkTable(ops.DatabaseTable):
+    pass
+
+
+class SparkClient(client.Client):
+
+    def __init__(self, dictionary):
+        conf = (SparkConf()
+                .setAppName('ibis')
+                .setMaster('local[*]'))
```

Comment:
Ultimately we probably want this to configurable since we likely don't always want to spin up local spark. Nothing to do here for now, just noting this.